iml-agent/src/action_plugins/lamigo.rs

```diff
@@ -0,0 +1,180 @@
+    /// File system name
+    fs: String,
+
+    #[structopt(long)]
+    /// MDT device that provides changelogs
+    mdt: String,
+
+    #[structopt(long)]
+    /// The user on which behalf lamigo consumes changelogs from the MDT device
+    user: String,
+
+    #[structopt(long)]
+    /// Fast OST pool name
+    hot_pool: String,
+
+    #[structopt(long)]
+    /// Slow OST pool name
+    cold_pool: String,
+
+    #[structopt(long)]
+    /// Interval for lamigo to wait before replicating a closed file (in seconds)
+    min_age: u32,
+
+    #[structopt(long)]
+    /// Lustre client mount point, e.g. `/mnt/lustre`
+    mount_point: String,
+
+    #[structopt(long)]
+    /// Lustre device to be mounted, e.g. `192.168.0.100@tcp0:/spfs`
+    lustre_device: String,
+
+    #[structopt(long)]
+    /// (Optional argument)
+    /// By default, nothing is performed, if the mount point is already mounted.
+    /// This flag is to force the mount execution.
+    force: bool,
+}
+
+impl fmt::Display for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "[Unit]\n\
+             Description=Run lamigo service\n\
+             [Service]\n\
+             ExecStart=/usr/bin/lamigo \
+             -m {fs}-{mdt} \
+             -u {user} \
+             -s {hot_pool} \
+             -t {cold_pool} \
+             -a {min_age} \
+             {mount_point}\n\
+             ",
+            fs = self.fs,
+            mdt = self.mdt,
+            cold_pool = self.cold_pool,
+            hot_pool = self.hot_pool,
+            min_age = self.min_age,
+            user = self.user,
+            mount_point = self.mount_point,
+        )
+    }
+}
+
+pub async fn create_lamigo_service_unit(c: Config) -> Result<(), ImlAgentError> {
+    if c.force {
+        create_lamigo_service_unit_internal(SYSTEMD_DIR, &c)
+            .err_into()
+            .await
+    } else {
+        let is_mounted = is_filesystem_mounted(&c.mount_point).await?;
+        if !is_mounted {
+            mount_filesystem(&c).await?;
+        }
+        create_lamigo_service_unit_internal(SYSTEMD_DIR, &c).await?;
+        Ok(())
+    }
+}
+
+pub async fn mount_filesystem(c: &Config) -> std::io::Result<()> {
+    // according to http://wiki.lustre.org/Mounting_a_Lustre_File_System_on_Client_Nodes
+    // we need to execute mount command
+    //   mount -t lustre \
+    //     [-o <options> ] \
+    //     <MGS NID>[:<MGS NID>]:/<fsname> \
+    //     /lustre/<fsname>
+    let process: Child = Command::new("mount")
+        .arg("-t")
+        .arg("lustre")
+        .arg(c.lustre_device.clone())
+        .arg(c.mount_point.clone())
+        .spawn()?;
+    let output: Output = process.wait_with_output().await?;
+    match output.status.success() {
+        true => Ok(()),
+        false => Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!(
+                "Cannot execute command \'mount -t lustre {} {}\'",
```

Comment:
`Command` has the `Debug` trait, maybe use it?

Reply:
Ah, this is easy, thanks!